src/libraries/System.Diagnostics.DiagnosticSource/tests/ActivitySourceTests.cs

```diff
@@ -881,6 +889,311 @@ public void RestoreOriginalParentTest()
+                listener.Sample = (ref ActivityCreationOptions<ActivityContext> activityOptions) => ActivitySamplingResult.AllDataAndRecorded;
+                ActivitySource.AddActivityListener(listener);
+
+                ActivityContext ac = new ActivityContext(ActivityTraceId.CreateRandom(), ActivitySpanId.CreateRandom(), ActivityTraceFlags.None, "key-value");
+                linksToCheckAgainst = new List<ActivityLink>() {new ActivityLink(ac)};
+
+                using (Activity a = aSource.CreateActivity("a", ActivityKind.Server, default(string), default, linksToCheckAgainst))
+                {
+                    Assert.NotNull(a);
+                    a.AddTag("tag1", "value1");
+                    a.AddBaggage("baggage1", "value1");
+
+                    tagsToCheckAgainst = new List<KeyValuePair<string, object>>() { new KeyValuePair<string, object>("tag1", "value1")};
+                    baggageToCheckAgainst = new List<KeyValuePair<string, string>>() { new KeyValuePair<string, string>("baggage1", "value1")};
+
+                    a.Start();
+                }
+            }).Dispose();
+        }
+
+        // --------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+        // Here is the sequence for how we determin the IdFormat of the Activity we create using ActivitySource:
+        // --------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+        //  - Use the IdFormat passed to the ActivitySource.Create if it is not "Unknown". Otherwise, goto next step.
+        //  - If Activity.ForceDefaultIdFormat is true, then use the value Activity.DefaultIdFormat.  Otherwise, goto next step.
+        //  - If the parentId string is null and parent ActivityContext is default and we have Activity.Current != null, then use Activity.Current.IdFormat. Otherwise, goto next step.
+        //  - If we have non default parent ActivityContext, then use W3C. Otherwise, goto next step.
+        //  - if we have non null parent id string, try to parse it to W3C. If parsing succeeded then use W3C format. otherwise use Hierarchical format.
+        // --------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+
+        //           -----------------------------------------------------------------------------------------------------------------------------------------------------------------
+        //           Default Id Format, Id fromat to force, Parent id format,   parent id,  default parent context, validate trace Id,  expected create format, expected start format
```

Comment:
```suggestion
        //           Activity.DefaultIdFormat, Activity.Create(idFormat), Activity.Current.IdFormat,   parent id,  default parent context, validate trace Id,  expected create format, expected start format
        //           (ForceIdFormat=true)
```
I got the wrong idea about what these columns meant until I reverse engineered it from the source. This might help clarify the intent.

Reply:
will do